cluster-autoscaler/core/scaledown/actuation/update_latency_tracker.go

```diff
@@ -146,3 +147,32 @@ func NewUpdateLatencyTrackerForTesting(nodeLister kubernetes.NodeLister, now fun
 	updateLatencyTracker.sleepDurationWhenPolling = time.Millisecond
 	return updateLatencyTracker
 }
+
+type latencyObservation struct {
+	latency    time.Duration
+	observedAt time.Time
+}
+
+func removeOldLatencies(latencies *list.List, expireTime time.Time) {
```

Comment:
Can you use [expiring.List](https://github.com/kubernetes/autoscaler/blob/34690b19c19b7779691fc8375e15494984586394/cluster-autoscaler/utils/expiring/list.go) for this? Right now they both implement essentially the same behavior.

Reply:
Updated